llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5997,7 +6023,7 @@ LoopVectorizationCostModel::getConsecutiveMemOpCost(Instruction *I,
          "Stride should be 1 or -1 for consecutive memory access");
   const Align Alignment = getLoadStoreAlignment(I);
   InstructionCost Cost = 0;
-  if (Legal->isMaskRequired(I)) {
+  if (isPredicatedInst(I)) {
```

Comment:
Are those changes still needed for the latest version?

Reply:
I think we'd better use this function, it has some extra checks for the predicated instructions (like dropping some uniform instructions)

Reply:
Possible, but this could be done separately (+ test showing  an improvement) if it is not needed to change where the styles are set?